lib/OnyxConnectionManager.ts

```diff
@@ -0,0 +1,266 @@
+import bindAll from 'lodash/bindAll';
+import * as Logger from './Logger';
+import type {ConnectOptions} from './Onyx';
+import OnyxUtils from './OnyxUtils';
+import * as Str from './Str';
+import type {DefaultConnectCallback, DefaultConnectOptions, OnyxKey, OnyxValue} from './types';
+
+type ConnectCallback = DefaultConnectCallback<OnyxKey>;
```

Comment:
Because `key` is defined as `TKey` on `DefaultConnectCallback` type, while `CollectionConnectCallback` has `key` as `undefined`, which creates some typing problems here.

Reply:
Same thought. Was this to simplify casting somewhere?

Reply:
> Because `key` is defined as `TKey` on `DefaultConnectCallback` type, while `CollectionConnectCallback` has `key` as `undefined`, which creates some typing problems here.

This brings me to [this comment](https://github.com/Expensify/react-native-onyx/pull/567#discussion_r1723116323) again... Why is it `undefined` in the first place?

Reply:
> Same thought. Was this to simplify casting somewhere?

Yes, exactly.

> This brings me to https://github.com/Expensify/react-native-onyx/pull/567#discussion_r1723116323 again... Why is it undefined in the first place?

Answered [here](https://github.com/Expensify/react-native-onyx/pull/567#discussion_r1723366610). But even with that changes I still need this type, using `DefaultConnectCallback` and `CollectionConnectCallback` leaves to type incompatibilities